packages/react-codemirror/src/e2e_tests/historyNavigation.spec.tsx

```diff
@@ -31,7 +31,37 @@ test('respects preloaded history', async ({ page, mount }) => {
   await expect(page.getByText(initialValue)).toBeVisible();
 });
 
-test('can execute queries and see them in history', async ({ page, mount }) => {
+test('can add new lines without onExecute', async ({ page, mount }) => {
+  const editorPage = new CypherEditorPage(page);
+
+  const editorComponent = await mount(<CypherEditor />);
+
+  // Ctrl-Enter does nothing when onExecute is false
+  await editorPage.getEditor().press('Control+Enter');
+  await expect(editorComponent).toHaveText('1\n', {
+    useInnerText: true,
+  });
+
+  // Enter adds new lines
+  await editorPage.getEditor().fill('Brock');
+  await editorPage.getEditor().press('Enter');
+  await editorPage.getEditor().press('Enter');
+  await expect(editorComponent).toHaveText('1\n2\n3\nBrock', {
+    useInnerText: true,
+  });
```

Comment:
I'm not understanding this test. So we are in the case `newLineOnEnter = false` here, which means a single line + enter should execute the query, which is not what this is doing.

Reply:
It is true that `newLineOnEnter = false`, but `onExecute` is also not set in this test so we should _never_ execute the query. So single line + enter should _not_ execute the query.

As for testing the case where `onExecute` is not set and `newLineOnEnter = true`, I considered that as well, but this case is equivalent to when `newLineOnEnter = false`. It doesn't matter whether it is true or false, the editor behaves the same when `onExecute` is false. I almost think it's better  to change the component prop type so it's not allowed to set `newLineOnEnter = true` if `onExecute = false`.

Reply:
Well my point is we are not testing that code. Your reasoning is correct, but because you know how the code is going to behave.

Reply:
I see what you are saying and I've added a test.

I still think it would be nice if we could prevent the user from setting `newLineOnEnter = true` if `onExecute = false` since this is arguably invalid, but I couldn't find a non-breaking way to enforce that with the TypeScript prop types.